aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -1018,6 +1072,100 @@ module aptos_framework::delegation_pool {
         };
     }
 
+    /// Allows a delegator to lock up their shares for at least as long as the specified Aptos governance proposal in
+    /// addition to the pool's unlock rules. This is useful for voting on proposals that require a longer lockup period
+    /// than the delegation pool's lockup period.
+    ///
+    /// If the pool's lockup or any existing extra lock the delegator has is already sufficient to cover the voting
+    /// duration of the governance proposal, no changes will be made.
+    public entry fun lock_shares(
```

Comment:
Is this practical? I think no body will call this in practice.

Reply:
This will likely be done by the community operators/voters